cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -181,4 +217,43 @@ class UploadResultToSw360Command : CliktCommand(
 
     private fun createReleaseName(pkgId: Identifier) =
         listOf(pkgId.namespace, pkgId.name).filter { it.isNotEmpty() }.joinToString("/")
+
+    private fun uploadAttachmentToRelease(
+        sw360ReleaseClient: SW360ReleaseClientAdapter,
+        release: SW360Release,
+        path: Path,
+        type: SW360AttachmentType
+    ): SW360Release? {
+        val build = AttachmentUploadRequest.builder(release)
+            .addAttachment(path, type)
+            .build()
+
+        val uploadAttachmentResponse = sw360ReleaseClient.uploadAttachments(build)
+
+        return if (uploadAttachmentResponse.isSuccess) {
+            uploadAttachmentResponse.target.also {
+                log.debug("Uploaded attachment ${path.fileName} with type $type to SW360.")
+            }
+        } else {
+            release.also {
+                log.error("Could not upload attachment: " + uploadAttachmentResponse.failedUploads())
+            }
+        }
+    }
+
+    private fun saveSourcesTemporarily(pkg: Package, tempDirectory: File): File? {
```

Comment:
This does more than just "saving sources temporarily". Also, we should probably not assume here that the caller is passing in a *temporary* directory; it could be just any directory.

In general I believe it makes sense to generalize the `DownloaderCommand`'s `archive(All)` functions (by returning a `File?`), moving them to a common place in the `cli` module, and making use of them here. Maybe you could pair with @MarcelBochtler for this?

Reply:
I'll do some prep-work for this.

Reply:
Here's a proposal: https://github.com/oss-review-toolkit/ort/pull/4098

Reply:
Yes i think it's better to generalize the archive funtion in a common place and both commands can use it.

Reply:
@sschuberth Can you check the new commit https://github.com/oss-review-toolkit/ort/pull/4094/commits/04082f32a50ed6f474f58a2f6517f48831e2592c and the reuse of it in the second one?